File: example/react/rest-client/src/__generated/model/entities/Book.ts
import type { Author, BookStore } from './';

export interface Book {
    
    readonly createdTime: string;
    
    readonly modifiedTime: string;
    
    readonly tenant: string;
    
    readonly id: number;
    
    readonly name: string;
    
    readonly edition: number;
    
    readonly price: number;
    
    readonly store?: BookStore;
    
    readonly authors: ReadonlyArray<Author>;
    
    readonly storeId?: number;
    
    readonly authorIds: ReadonlyArray<number>;
}

File: project/jimmer-spring-boot-starter/src/test/java/org/babyfish/jimmer/spring/java/model/Book.java
package org.babyfish.jimmer.spring.java.model;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Entity
public interface Book {

    @Id
    UUID id();

    String name();

    int edition();

    BigDecimal price();

    /**
     * The bookstore to which the current book belongs, null is allowed
     */
    @ManyToOne
    @Nullable
    BookStore store();

    /**
     * All authors involved in writing the work
     */
    @ManyToMany
    @JoinTable(deletedWhenEndpointIsLogicallyDeleted = true)
    List<Author> authors();
}
